### PR TITLE
modified documentation for customize identity model

### PR DESCRIPTION
**Pull request by fiyazbinhasan**
Fixes https://github.com/dotnet/AspNetCore.Docs/issues/22179

@Rick-Anderson I did a lot of modifications. I do think that at this point we don't even need the `moniker range=">= aspnetcore-2.1 <= aspnetcore-5.0"` since I removed everything for `1.1` and `2.0`. And from 2.1 to 5.0 the snippets are all same. 

Please review.

### CODE REVIEW

File: aspnetcore/security/authentication/customize-identity-model.md [hunk 1]
```diff
@@ -342,15 +342,37 @@ Update *Pages/Shared/_LoginPartial.cshtml* and replace `IdentityUser` with `Appl
...
+    o.Stores.MaxLengthForKeys = 128;
+    o.SignIn.RequireConfirmedAccount = true;
+})
+.AddDefaultUI()
+.AddDefaultTokenProviders();
+```                
+
+::: moniker-end
+
+In ASP.NET Core 2.1 or later, Identity is provided as a Razor Class Library. For more information, see <xref:security/authentication/scaffold-identity>. Consequently, the preceding code requires a call to <xref:Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI\*>. If the Identity scaffolder was used to add Identity files to the project, remove the call to `AddDefaultUI`. For more information, see:
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/customize-identity-model.md:375]:
```suggestion
Identity is provided as a Razor Class Library. For more information, see <xref:security/authentication/scaffold-identity>. Consequently, the preceding code requires a call to <xref:Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI\*>. If the Identity scaffolder was used to add Identity files to the project, remove the call to `AddDefaultUI`. For more information, see:
```

**Review by Rick-Anderson — COMMENTED** [review]
> Fixes #22179
> 
> @Rick-Anderson I did a lot of modifications. I do think that at this point we don't even need the `moniker range=">= aspnetcore-2.1 <= aspnetcore-5.0"` since I removed everything for `1.1` and `2.0`. And from 2.1 to 5.0 the snippets are all same.
> 
> Please review.

Remove all the monikers, we only support 2.1 and later.

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]
remove monikers too

File: aspnetcore/security/authentication/customize-identity-model.md (hunk 1 above)
Comment (fiyazbinhasan) [reply aspnetcore/security/authentication/customize-identity-model.md]: resolved and removed monikers

File: aspnetcore/security/authentication/customize-identity-model.md [hunk 2]
```diff
@@ -342,15 +342,37 @@ Update *Pages/Shared/_LoginPartial.cshtml* and replace `IdentityUser` with `Appl
 @inject UserManager<ApplicationUser> UserManager
 ```
 
-Update *Areas/Identity/IdentityHostingStartup.cs*  or `Startup.ConfigureServices` and replace `IdentityUser` with `ApplicationUser`.
+Update *Areas/Identity/IdentityHostingStartup.cs* or `Startup.ConfigureServices` and replace `IdentityUser` with `ApplicationUser`.
+
+::: moniker range=">= aspnetcore-2.1 <= aspnetcore-5.0"
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/customize-identity-model.md:347]:
```suggestion
```

Remove moniker unless this doesn't work for 6. If 6 is different, you'd need a moniker for 6

File: aspnetcore/security/authentication/customize-identity-model.md [hunk 3]
```diff
@@ -342,15 +342,37 @@ Update *Pages/Shared/_LoginPartial.cshtml* and replace `IdentityUser` with `Appl
...
 
 ```csharp
-services.AddIdentity<ApplicationUser>()
-        .AddEntityFrameworkStores<ApplicationDbContext>()
-        .AddDefaultUI();
+services.AddDefaultIdentity<ApplicationUser>(options => options.SignIn.RequireConfirmedAccount = true)
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/customize-identity-model.md:350]: Why isn't this

```suggestion
services.AddDefaultIdentity<IdentityUser>(options => options.SignIn.RequireConfirmedAccount = true)
```

File: aspnetcore/security/authentication/customize-identity-model.md [hunk 4]
```diff
@@ -342,15 +342,37 @@ Update *Pages/Shared/_LoginPartial.cshtml* and replace `IdentityUser` with `Appl
...
+services.AddIdentityCore<TUser>(o =>
+{
+    o.Stores.MaxLengthForKeys = 128;
+    o.SignIn.RequireConfirmedAccount = true;
+})
+.AddDefaultUI()
+.AddDefaultTokenProviders();
+```                
+
+::: moniker-end
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/customize-identity-model.md:373]:
```suggestion
```

File: aspnetcore/security/authentication/customize-identity-model.md [hunk 5]
```diff
@@ -342,15 +342,37 @@ Update *Pages/Shared/_LoginPartial.cshtml* and replace `IdentityUser` with `Appl
...
 ```csharp
-services.AddIdentity<ApplicationUser>()
-        .AddEntityFrameworkStores<ApplicationDbContext>()
-        .AddDefaultUI();
+services.AddDefaultIdentity<ApplicationUser>(options => options.SignIn.RequireConfirmedAccount = true)
+        .AddEntityFrameworkStores<ApplicationDbContext>();                                    
 ```
 
-In ASP.NET Core 2.1 or later, Identity is provided as a Razor Class Library. For more information, see <xref:security/authentication/scaffold-identity>. Consequently, the preceding code requires a call to <xref:Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI*>. If the Identity scaffolder was used to add Identity files to the project, remove the call to `AddDefaultUI`. For more information, see:
+Calling `AddDefaultIdentity<TUser>()` is equivalent to the following,
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/customize-identity-model.md:354]: Might be nice to link to source where they can see this.

File: aspnetcore/security/authentication/customize-identity-model.md [hunk 6]
```diff
@@ -364,172 +386,70 @@ Follow these steps to change the PK type:
...
-    [!code-csharp[](customize-identity-model/samples/1.1/MvcSampleApp/Data/ApplicationDbContext.cs?name=snippet_ApplicationDbContext&highlight=5-6)]
-
-    Register the custom database context class when adding the Identity service in `Startup.ConfigureServices`:
-
-    [!code-csharp[](customize-identity-model/samples/1.1/MvcSampleApp/Startup.cs?name=snippet_ConfigureServices&highlight=7-9)]
-
-    The <xref:Microsoft.Extensions.DependencyInjection.IdentityEntityFrameworkBuilderExtensions.AddEntityFrameworkStores*> method accepts a `TKey` type indicating the primary key's data type.
-
-    ::: moniker-end
+   In ASP.NET Core 2.1 or later, Identity is provided as a Razor Class Library. For more information, see <xref:security/authentication/scaffold-identity>. Consequently, the preceding code requires a call to <xref:Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI%2A>. If the Identity scaffolder was used to add Identity files to the project, remove the call to `AddDefaultUI`.
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/customize-identity-model.md:452]:
```suggestion
  Identity is provided as a Razor Class Library. For more information, see <xref:security/authentication/scaffold-identity>. Consequently, the preceding code requires a call to <xref:Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI%2A>. If the Identity scaffolder was used to add Identity files to the project, remove the call to `AddDefaultUI`.
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/security/authentication/customize-identity-model.md (hunk 3 above)
Comment (fiyazbinhasan) [reply aspnetcore/security/authentication/customize-identity-model.md:350]: In this step, the doc suggests that we replace IdentityUser with the user-defined ApplicationUser type

File: aspnetcore/security/authentication/customize-identity-model.md (hunk 2 above)
Comment (fiyazbinhasan) [reply aspnetcore/security/authentication/customize-identity-model.md]: resolved

File: aspnetcore/security/authentication/customize-identity-model.md (hunk 4 above)
Comment (fiyazbinhasan) [reply aspnetcore/security/authentication/customize-identity-model.md]: resolved

File: aspnetcore/security/authentication/customize-identity-model.md (hunk 5 above)
Comment (fiyazbinhasan) [reply aspnetcore/security/authentication/customize-identity-model.md]: Added link to the API doc

File: aspnetcore/security/authentication/customize-identity-model.md (hunk 6 above)
Comment (fiyazbinhasan) [reply aspnetcore/security/authentication/customize-identity-model.md]: resolved

**Comment by Rick-Anderson** [comment]
Here's a tip for next time. All the suggestions that you plan to accept, select "add to batch". When you're done adding them, commit. No need to accept and reply.

image

**Comment by fiyazbinhasan** [comment]
Thanks for the tip! 🙂

File: aspnetcore/security/authentication/customize-identity-model.md [hunk 7]
```diff
@@ -342,15 +342,33 @@ Update *Pages/Shared/_LoginPartial.cshtml* and replace `IdentityUser` with `Appl
...
 ```csharp
-services.AddIdentity<ApplicationUser>()
-        .AddEntityFrameworkStores<ApplicationDbContext>()
-        .AddDefaultUI();
+services.AddDefaultIdentity<ApplicationUser>(options => options.SignIn.RequireConfirmedAccount = true)
+        .AddEntityFrameworkStores<ApplicationDbContext>();                                    
 ```
 
-In ASP.NET Core 2.1 or later, Identity is provided as a Razor Class Library. For more information, see <xref:security/authentication/scaffold-identity>. Consequently, the preceding code requires a call to <xref:Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI*>. If the Identity scaffolder was used to add Identity files to the project, remove the call to `AddDefaultUI`. For more information, see:
+Calling <xref:Microsoft.Extensions.DependencyInjection.IdentityServiceCollectionUIExtensions.AddDefaultIdentity%2A> is equivalent to the following,
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/customize-identity-model.md:352]:
```suggestion
Calling <xref:Microsoft.Extensions.DependencyInjection.IdentityServiceCollectionUIExtensions.AddDefaultIdentity%2A> is equivalent to the following code:
```

**Comment by Rick-Anderson** [comment]
Nice work @fiyazbinhasan 
You're welcome to take any issue, even those assigned to me or anyone else. Let me know if you'd like me to recommend some for you.

**Comment by fiyazbinhasan** [comment]
@Rick-Anderson would be great if you can recommend me some or I can pick some on my own. :smiley:

**Comment by Rick-Anderson** [comment]
I'll look for some tomorrow.

**Pull request merged**